ctest-next/templates/test.rs

```diff
@@ -137,6 +141,95 @@ mod generated_tests {
     }
 {%- endfor +%}
+
+{%- for item in ctx.roundtrip_tests +%}
+
+    /// Generates a padding map for a specific type.
+    ///
+    /// Essentially, it returns a list of bytes, whose length is equal to the size of the type in
+    /// bytes. Each element corresponds to a byte and has two values. `true` if the byte is padding,
+    /// and `false` if the byte is not padding.
+    ///
+    /// For type aliases, the padding map is all zeroes.
+    fn roundtrip_padding__{{ item.id }}() -> Vec<bool> {
+        {%- if item.fields.len() == 0 +%}
+        vec![!{{ item.is_alias }}; size_of::<{{ item.id }}>()]
+        {%- else +%}
+        let mut v = Vec::<(usize, usize)>::new();
+        let bar = MaybeUninit::<{{ item.id }}>::zeroed();
+        let bar = bar.as_ptr();
+        {%- for field in item.fields +%}
+        unsafe {
+            let ty_ptr = &raw const ((*bar).{{ field.ident() }});
+            let val = ty_ptr.read_unaligned();
+            let size = size_of_val(&val);
+            let off = offset_of!({{ item.id }}, {{ field.ident() }});
+            v.push((off, size));
+        }
+        {%- endfor +%}
+        // This vector contains `true` if the byte is padding
+        // and `false` if the byte is not padding.
+        // Initialize all bytes as:
+        //  - padding if we have fields, this means that only
+        //  the fields will be checked
+        //  - no-padding if we have a type alias: if this
+        //  causes problems the type alias should be skipped
+        let mut pad = vec![true; size_of::<{{ item.id }}>()];
+        for (off, size) in &v {
+            for i in 0..*size {
+                pad[off + i] = false;
+            }
+        }
+        pad
+        {%- endif +%}
+    }
+
+    /// Tests whether the type alias `x` when passed to C and back to Rust remains unchanged.
+    ///
+    /// It checks if the size is the same as well as if the padding bytes are all in the
+    /// correct place.
+    pub fn {{ item.test_name }}() {
+        type U = {{ item.id }};
+        extern "C" {
+            fn ctest_roundtrip__{{ item.id }}(
+                size: i32, x: MaybeUninit<U>, e: *mut c_int, pad: *const bool
+            ) -> U;
+        }
+        let pad = roundtrip_padding__{{ item.id }}();
+        unsafe {
+            let mut error: c_int = 0;
+            let mut y = MaybeUninit::<U>::zeroed();
+            let mut x = MaybeUninit::<U>::zeroed();
+            let x_ptr = x.as_mut_ptr().cast::<u8>();
+            let y_ptr = y.as_mut_ptr().cast::<u8>();
+            let sz = size_of::<U>();
+            for i in 0..sz {
+                let c: u8 = (i % 256) as u8;
+                let c = if c == 0 { 42 } else { c };
+                let d: u8 = 255_u8 - (i % 256) as u8;
+                let d = if d == 0 { 42 } else { d };
+                x_ptr.add(i).write_volatile(c);
+                y_ptr.add(i).write_volatile(d);
+            }
```

Comment:
Could you add a comment? I think it's basically filling up the bytes with a deterministic pattern to detect when those unintentionally change, one pattern will be expected on the C side and the other will be expected on the Rust side